src/main.rs

```diff
@@ -10,53 +11,54 @@ fn get_filenames() -> (String, String) {
     (files[1].clone(), files[2].clone())
 }
 
+/// Reads the given file and outputs it line by line.
 fn read_file_lines(filename: String) -> Vec<String> {
     let file: String = fs::read_to_string(filename).expect("Couldn't read file ");
     file.lines().map(|l| l.to_string()).collect()
 }
 
-fn lcs(f1: &[String], f2: &[String]) -> Vec<Vec<usize>> {
-    let m = &f1.len();
-    let n = &f2.len();
+/// Builds a grid with the longest common subsequence algorithm.
+fn longest_common_subsequence(file1: &[String], file2: &[String]) -> Vec<Vec<usize>> {
+    let file1_len = &file1.len();
+    let file2_len = &file2.len();
 
-    let mut c = vec![vec![0; *n + 1]; *m + 1];
+    let mut result_grid = vec![vec![0; *file2_len + 1]; *file1_len + 1];
 
-    for (i, f1_line) in f1.iter().enumerate() {
-        for (j, f2_line) in f2.iter().enumerate() {
+    for (i, f1_line) in file1.iter().enumerate() {
+        for (j, f2_line) in file2.iter().enumerate() {
             if f1_line == f2_line {
-                c[i + 1][j + 1] = c[i][j] + 1;
+                result_grid[i + 1][j + 1] = result_grid[i][j] + 1;
             } else {
-                c[i + 1][j + 1] = cmp::max(c[i + 1][j], c[i][j + 1])
+                result_grid[i + 1][j + 1] = cmp::max(result_grid[i + 1][j], result_grid[i][j + 1])
             }
         }
     }
 
-    c
+    result_grid
 }
 
-fn print_diff(c: &Vec<Vec<usize>>, f1: &[String], f2: &[String], i: usize, j: usize) {
-    match (i, j) {
-        (i, j) if i > 0 && j > 0 && f1[i - 1] == f2[j - 1] => {
-            print_diff(c, f1, f2, i - 1, j - 1);
-            println!("{}", f1[i - 1]);
-        },
-        (i, j) if j > 0 && (i == 0 || c[i][j - 1] >= c[i - 1][j]) => {
-            print_diff(c, f1, f2, i, j - 1);
-            println!("> {}", f2[j - 1]);
-        },
-        (i, j) if i > 0 && (j == 0 || c[i][j - 1] < c[i - 1][j]) => {
-            print_diff(c, f1, f2, i - 1, j);
-            println!("< {}", f1[i - 1]);
-        },
-        _ => println!(),
-    };
+/// Prints out the difference between the two given vectors of strings, using their previously
+/// calculated largest common subsequence grid.
+fn print_diff(subsequence_coparison_grid: &Vec<Vec<usize>>, file1: &[String], file2: &[String], file1_iter: usize, file2_iter: usize) {
```

Comment:
It says "coparison", it's "comparison". Please change it in all instances.

Reply:
Fixed in commit [c9e5d9a](https://github.com/lambdaclass/diff-in-rust-jm/pull/4/commits/c9e5d9ab3fba23ccb178a6b4a06a0800f6814e2d)